src/index.js

```diff
@@ -1,40 +1,15 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import ReactDOMServer from 'react-dom/server';
-import hypernova, { serialize, load } from 'hypernova';
+import hypernova from 'hypernova';
+import { renderReactClient } from './client';
+import { renderReactServer, renderReactStaticServer } from './server';
 
 export const renderReact = (name, component) => hypernova({
-  server() {
-    return (props) => {
-      const contents = ReactDOMServer.renderToString(React.createElement(component, props));
-      return serialize(name, contents, props);
-    };
-  },
+  server: () => renderReactServer(name, component),
 
-  client() {
-    const payloads = load(name);
-
-    if (payloads) {
-      payloads.forEach((payload) => {
-        const { node, data } = payload;
-        const element = React.createElement(component, data);
-
-        if (ReactDOM.hydrate) {
-          ReactDOM.hydrate(element, node);
-        } else {
-          ReactDOM.render(element, node);
-        }
-      });
-    }
-
-    return component;
-  },
+  client: () => renderReactClient(name, component),
 });
 
 export const renderReactStatic = (name, component) => hypernova({
-  server() {
-    return props => ReactDOMServer.renderToStaticMarkup(React.createElement(component, props));
-  },
+  server: () => renderReactStaticServer(name, component),
 
-  client() {},
```

Comment:
nit: `client() {}` is preferred over `client: () => {}`

Reply:
thanks for noticing, arrow function is indeed not necessary in object methods